repo_policy_compliance/github_client.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for GitHub client."""
+
+import functools
+import os
+from typing import Callable, Concatenate, ParamSpec, TypeVar
+
+from github import Github
+
+from .exceptions import InputError
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+# Bandit thinks this constant is the real Github token
+GITHUB_TOKEN_ENV_NAME = "GITHUB_TOKEN"  # nosec
+
+
+def inject(func: Callable[Concatenate[Github, P], R]) -> Callable[P, R]:
+    """Injects a GitHub client as the first argument to a function.
+
+    Args:
+        func: The function that requires a GitHub client.
+
+    Returns:
+        The function where the GitHub client has been injected as the first argument.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
+        """Replace function.
+
+        Args:
+            args: The positional arguments passed to the method, github_client is prepended when
+            calling the wrapped method
```

Comment:
Should this be indented by one more level?

Reply:
Fixed